src/main/kotlin/no/nav/syfo/service/AktivitetspliktForhandsvarselVarselService.kt

```diff
@@ -0,0 +1,65 @@
+package no.nav.syfo.service
+
+import no.nav.syfo.BRUKERNOTIFIKASJON_AKTIVITETSKRAV_FORHANDSVARSEL_STANS_TEXT
+import no.nav.syfo.consumer.distribuerjournalpost.DistibusjonsType
+import no.nav.syfo.kafka.consumers.varselbus.domain.ArbeidstakerHendelse
+import no.nav.syfo.kafka.consumers.varselbus.domain.VarselData
+import no.nav.syfo.kafka.consumers.varselbus.domain.toVarselData
+import no.nav.syfo.kafka.producers.brukernotifikasjoner.BrukernotifikasjonKafkaProducer
+import org.apache.commons.cli.MissingArgumentException
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.net.URL
+import java.util.*
+
+class AktivitetspliktForhandsvarselVarselService(
+    val senderFacade: SenderFacade,
+    val accessControlService: AccessControlService,
+    val journalpostPageUrl: String,
+    private val isSendingEnabled: Boolean,
+) {
+    fun sendVarselTilArbeidstaker(varselHendelse: ArbeidstakerHendelse) {
+        if (isSendingEnabled) {
+            log.info("[FORHAANDSVARSEL] sending enabled")
+            val userAccessStatus = accessControlService.getUserAccessStatus(varselHendelse.arbeidstakerFnr)
+            val varselData = dataToVarselData(varselHendelse.data)
+            requireNotNull(varselData.journalpost)
+            requireNotNull(varselData.journalpost.id)
+
+            if (userAccessStatus.canUserBeDigitallyNotified) {
+                log.info("Sending [FORHAANDSVARSEL] to brukernotifikasjoner")
+                senderFacade.sendTilBrukernotifikasjoner(
+                    uuid = varselData.journalpost.uuid,
+                    mottakerFnr = varselHendelse.arbeidstakerFnr,
+                    content = BRUKERNOTIFIKASJON_AKTIVITETSKRAV_FORHANDSVARSEL_STANS_TEXT,
+                    url = URL("$journalpostPageUrl/${varselData.journalpost.id}"),
+                    varselHendelse = varselHendelse,
+                    meldingType = BrukernotifikasjonKafkaProducer.MeldingType.BESKJED,
```

Comment:
Denne er visst noe lovkrav om at må være oppgave, pga revarsling osv.

Men i så fall må vi ha på plass ferdigstilling tror jeg, fordi journalpost-siden ikke vil kunne ferdigstille.
Litt usikker på hvordan best løse det, det må kanskje enten bli
- lenke til vår infoside som må ha visning av forhåndsvarsel på plass, samt ferdigstilling
- lage en proxy-redirect-url i for eksempel esyfo-proxy, som ferdigstiller varsel og redirecter videre. (litt hackete løsning kanskje).

Reply:
@AudunSorheim Okey, fint du sa 👍 Tror likevel det blir en TODO her. Ekstern varsling for aktivitetskrav er skrudd av i prod for nå (burde nevnt det i beskrivelsen over), og vil ikke komme på før del 2 (ref muralen)

Reply:
Ja :) Så det går nok fint

Reply:
@AudunSorheim Lager en TODO 👍